src/nodes.coffee

```diff
@@ -1104,47 +1115,77 @@ exports.Obj = class Obj extends Base
 
   children: ['properties']
 
+  isAssignable: ->
+    @lhs = true
```

Comment:
I don’t see why you think `lhs` is hacky; surely it’s better to isolate this object-specific property to the `Obj` class? Seems okay to me.

I think we should probably tackle destructured function parameters in this same PR, because basically we will need to be adding special cases to the same places you just modified in this PR to cover the things that you can normally destructure but not in a function parameter list, like `this`. I understand the appeal of merging this in and then using that as the new baseline, and only seeing the changes related to function parameters in a new PR, but since the function-parameters work is so related to this work I think it would be more useful to still see these changes. I don’t feel strongly though. There’s also https://github.com/geoffreybooth/coffeescript/pull/4 to merge in.